compiler/destroyer.nim

```diff
@@ -601,6 +601,12 @@ proc p(n: PNode; c: var Con): PNode =
   of nkNone..nkNilLit, nkTypeSection, nkProcDef, nkConverterDef, nkMethodDef,
       nkIteratorDef, nkMacroDef, nkTemplateDef, nkLambda, nkDo, nkFuncDef:
     result = n
+  of nkCast, nkHiddenStdConv, nkHiddenSubConv, nkConv:
```

Comment:
Are you sure these node kinds always have 2 children? I remember them being inconsistent...

Reply:
> Are you sure these node kinds always have 2 children?

Yeah, a quick `grep` shows they do. The last three are usually handled together (see `genConv`) and there's a test case for the `nkCast` case.